src/stores/ProofStore.ts

```diff
@@ -1,126 +1,72 @@
+import { ContractReceipt } from 'ethers'
+import { DataKeys } from 'models/DataKeys'
 import { PersistableStore } from '@big-whale-labs/stores'
-import {
-  getEddsaPublicKey,
-  requestAddressOwnershipAttestation,
-  requestBalanceAttestation,
-} from 'helpers/proofs/attestor'
+import { Web3Provider } from '@ethersproject/providers'
 import { proxy } from 'valtio'
-import BaseProof from 'helpers/proofs/BaseProof'
-import ERC721Proof, { ERC721ProofSchema } from 'helpers/proofs/ERC721Proof'
-import EmailProof, { EmailProofSchema } from 'helpers/proofs/EmailProof'
-import Network from 'models/Network'
-import ProofResult from 'models/ProofResult'
+import Proof from 'models/Proof'
 import WalletStore from 'stores/WalletStore'
-import checkNavigator from 'helpers/proofs/checkNavigator'
 import dataShapeObject from 'helpers/contracts/dataShapeObject'
 import env from 'helpers/env'
-import getNullifierMessage from 'helpers/proofs/getNullifierMessage'
 import handleError from 'helpers/handleError'
 
-class ProofStore extends PersistableStore {
-  proofsCompleted: BaseProof[] = []
+export class ProofStore extends PersistableStore {
+  proofsCompleted: Proof[] = []
+  dataKey: DataKeys
+  createBadge: (
+    provider: Web3Provider,
+    proof: Proof
+  ) => Promise<ContractReceipt>
 
-  deleteProof(proof: BaseProof) {
-    this.proofsCompleted = this.proofsCompleted.filter((p) => !proof.equal(p))
+  constructor(
+    dataKey: DataKeys,
```

Comment:
Maybe `dataKeys`? If I try to use this store I will think, that I have only one key, and I'll try to find, why I have only one key, why we need only one key, why don't we store all keys, or do we? So many questions because of the naming